### PR TITLE
Devices API docs

### PR DESCRIPTION
**Pull request by apoorvadeshpande-okta**
## Description:
- **What's changed?** 

New API documentation for Devices

- **Is this PR related to a Monolith release?** 

Yes. TBD

### Resolves:

* [OKTA-252194](https://oktainc.atlassian.net/browse/OKTA-252194)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 1]
```diff
@@ -0,0 +1,2207 @@
...
+# Device API
+
+<ApiLifecycle access="ea" />
+
+The Okta Device API provides centralized integration platform to store and manage device information. Okta administrators can use these APIs to manage workfoce identity device object information.
+
+The Device API supports the following **Device Operations**:
+* Create, Get, Update, Delete device objects
+* Perform lifecycle transitions on the device objects
+* Link devices to the devices and manage these associations
```
Comment (brennerryan-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: Is this a typo? I'm wondering if it is supposed to say "link users to the devices"

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 2]
```diff
@@ -0,0 +1,2207 @@
...
+* [Activate](#activate-device)
+* [Deactivate](#deactivate-device)
+* [Suspend](#suspend-device)
+* [Unsuspend](#unsuspend-device)
+
+##### Device User link operations
+
+* [Get All Users Linked to a Device](#get-all-users-linked-to-a-device)
+* [Get a User Linked to a Device](#get-a-user-linked-to-a-device)
+* [Get All Devices Linked to a User](#get-all-devices-linked-to-a-user)
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: ~Get All~ List Devices

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 3]
```diff
@@ -0,0 +1,2207 @@
...
+
+Fetches a device by its `id`. If you don't know the device `id`, you can [list Devices](#list-devices).
+
+Populated links indicate allowed actions on the device in the current status.
+
+#### Request Path Parameters
+
+| Parameter   | Type   | Description                      |
+| ----------- | ------ | -------------------------------- |
+| `id` | String | The device's unique identifier, generated by Okta |
```
[the remark this answers is not in the document]
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: `id` reflects attribute in the device object

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 4]
```diff
@@ -0,0 +1,2207 @@
...
+Fetches a list of all available devices for your org. Response will be paginated with maximum size of 200 devices.
+
+A subset of devices can be returned that match a supported search criteria using `search` query parameter.
+
+##### Request Parameters
+
+- [List All Devices](#list-all-devices) (no parameters)
+- [List Devices with Search](#list-devices-with-search) (`search`)
+
+| Parameter   | Description                                                                                                                                    | Param Type   | DataType   | Required |
```
[the remark this answers is not in the document]
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: After Friday's discussions I have changed this everywhere

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: changed to `deviceId` => The `id` of device object

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 5]
```diff
@@ -0,0 +1,2323 @@
...
+                ]
+            }
+        }
+    }
+}
+```
+
+### Update Device with PATCH
+
+<ApiOperation method="patch" url="/api/v1/devices/${deviceId}" />
```
Comment (jakubvul-okta) [inline packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: #765 adds support for this. We'll want to rebase from `master` once that PR is merged.

File: packages/@okta/vuepress-site/docs/reference/api/devices/index.md [hunk 6]
```diff
@@ -0,0 +1,2207 @@
...
+
+List devices with a platform of `MACOS`
+
+    search=profile.platform eq "MACOS"
+
+List devices with displayName starting with of `Eng-dev` that were created before `01/01/2014` or have a status of `ACTIVE`.
+
+    search=profile.displayName sw "Eng-dev" and (created lt "2014-01-01T00:00:00.000Z" or status eq "ACTIVE")
+
+##### Request Example
```
[the remark this answers is not in the document]
Comment (apoorvadeshpande-okta) [reply packages/@okta/vuepress-site/docs/reference/api/devices/index.md]: done

**Comment by apoorvadeshpande-okta** [comment]
Closing the PR for now. I will reopen this as needed.

**Pull request closed without merging**